qualang_tools/config/waveform_tools.py

```diff
@@ -26,46 +26,24 @@ def drag_gaussian_pulse_waveforms(
-        gauss_wave = amplitude * np.exp(-((t - center) ** 2) / (2 * sigma**2))  # The gaussian function
-        gauss_der_wave = (
-            amplitude * (-2 * 1e9 * (t - center) / (2 * sigma**2)) * np.exp(-((t - center) ** 2) / (2 * sigma**2))
-        )  # The derivative of gaussian
-        if subtracted:
-            gauss_wave = gauss_wave - gauss_wave[-1]  # subtracted gaussian
-        z = gauss_wave + 1j * 0
-        if alpha != 0:
-            # The complex DRAG envelope:
-            z += 1j * gauss_der_wave * (alpha / (delta - 2 * np.pi * detuning))
-            # The complex detuned DRAG envelope:
-            z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
-    else:
-        if alpha != 0 and anharmonicity == 0:
-            raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
-        t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
-        center = (length - 1e9 / sampling_rate) / 2
-        gauss_wave = amplitude * np.exp(-((t - center) ** 2) / (2 * sigma**2))  # The gaussian function
-        gauss_der_wave = (
-            amplitude * (-2 * 1e9 * (t - center) / (2 * sigma**2)) * np.exp(-((t - center) ** 2) / (2 * sigma**2))
-        )  # The derivative of gaussian
-        if subtracted:
-            gauss_wave = gauss_wave - gauss_wave[-1]  # subtracted gaussian
-        z = gauss_wave + 1j * 0
-        if alpha != 0:
-            # The complex DRAG envelope:
-            z += 1j * gauss_der_wave * (alpha / (2 * np.pi * anharmonicity - 2 * np.pi * detuning))
-            # The complex detuned DRAG envelope:
-            z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
+    if alpha != 0 and anharmonicity == 0:
+        raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
+    t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
+    center = (length - 1e9 / sampling_rate) / 2
+    gauss_wave = amplitude * np.exp(-((t - center) ** 2) / (2 * sigma**2))  # The gaussian function
+    gauss_der_wave = (
+        amplitude * (-2 * 1e9 * (t - center) / (2 * sigma**2)) * np.exp(-((t - center) ** 2) / (2 * sigma**2))
+    )  # The derivative of gaussian
+    if subtracted:
+        gauss_wave = gauss_wave - gauss_wave[-1]  # subtracted gaussian
+    z = gauss_wave + 1j * 0
+    if anharmonicity != detuning:
+        # The complex DRAG envelope:
+        z += 1j * gauss_der_wave * (alpha / (2 * np.pi * anharmonicity - 2 * np.pi * detuning))
+    # The complex detuned DRAG envelope:
+    z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
```

Comment:
You moved this outside of the if statement, it's ok because if `detuning=0` then it doesn't matter?

Reply:
yes exactly and detuning can be 0 no matter the value of alpha